libs/ml/include/ml/ops/strided_slice.hpp

```diff
@@ -0,0 +1,156 @@
+#pragma once
+//------------------------------------------------------------------------------
+//
+//   Copyright 2018-2019 Fetch.AI Limited
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+//------------------------------------------------------------------------------
+
+#include "ml/exceptions/exceptions.hpp"
+#include "ml/ops/ops.hpp"
+
+#include <cassert>
+#include <vector>
+
+namespace fetch {
+namespace ml {
+namespace ops {
+
+template <class T>
+class StridedSlice : public Ops<T>
+{
+public:
+  using TensorType    = T;
+  using SizeType      = fetch::math::SizeType;
+  using SizeVector    = fetch::math::SizeVector;
+  using ArrayPtrType  = std::shared_ptr<TensorType>;
+  using VecTensorType = typename Ops<T>::VecTensorType;
+  using SPType        = OpStridedSliceSaveableParams<T>;
+  using MyType        = StridedSlice<TensorType>;
+
+  explicit StridedSlice(SizeVector const &begins, SizeVector const &ends,
+                        SizeVector const &strides = {})
```

Comment:
it actually seems a little strange that this is a separate op from Slice() at all - it's calling the same implementation. I know that's how it is in tensorflow though so nothing to change here